### DISCUSSION TITLE
MP模式下重新组包的逻辑流程

### DISCUSSION THREAD

**Discussion by sharewzy** [Tencent/tquic#354] [General]
接收端和协议里面没看到太多和组包相关的流程，这部分现在大概的实现逻辑是什么样的。

**Comment by iyangsj** [comment]
可以参见源码关键部分： `recv()` -> `recv_packet()` -> `recv_frame()`
- https://github.com/Tencent/tquic/blob/develop/src/connection/connection.rs#L399
- https://github.com/Tencent/tquic/blob/develop/src/connection/connection.rs#L444
- https://github.com/Tencent/tquic/blob/develop/src/connection/connection.rs#L700

**Reply by sharewzy** [reply]
这里是按照PN来重组包吗？这部分代码我check过，但是还是没有太理解。

**Reply by iyangsj** [reply]
区分不同类型的QUIC帧，进行不同的处理。 例如Stream帧（携带应用层数据），按Stream帧中的offset/length 对数据块去重及合并
